### ISSUE TITLE
Fail to use secret-version latest

### ISSUE THREAD

**Issue by gogovan-vincentngai** [argoproj-labs/argocd-vault-plugin#406]
**Describe the bug**
secret-version: "latest" unable to use in version **1.12.0** for both iniline and annotations
This implementation is working fine in version **0.7.0**
But from the doc it should be able to use
https://argocd-vault-plugin.readthedocs.io/en/stable/backends/#versioned-secrets

**To Reproduce**
Vault Version 1.7.0
ArgoCD version v2.2.5+8f981cc
argocd-vault-plugin 1.12.0

I upgrade plugin from 0.7.0 to 1.12.0 inject the vault plugin via init container 
When i inject below from 
```
annotations:
    avp.kubernetes.io/path: "xxxx/data/xxxx"
    avp.kubernetes.io/secret-version: "latest"
```
it will return  error  saying
https://xxxxxx/v1/xxxx/data/xxxx?version=latest  return 500

However if i inject a specific version it will success like this
```
    avp.kubernetes.io/secret-version: "19"
```
I also did for iniline it will have same error

```
apiVersion: v1
kind: ConfigMap
metadata:
  name: test-cm2
data:
  hello: <path:xxxxxx/data/xxxxx#mykey#latest>
```

**Expected behavior**
secret-version latest version can be use in bot inline and annotations

**Screen Capture**

<img width="1261" alt="螢幕截圖 2022-09-13 上午6 24 26" src="https://user-images.githubusercontent.com/35673973/189819127-c9338f9a-f925-4c46-92fa-cdc292a0aaa5.png">

**Comment by erkanzileli** [comment]
Hi, I inspected the code and it looks fine. Let me explain it briefly.

If you look at the Vault client code in version [0.7.0](https://github.com/argoproj-labs/argocd-vault-plugin/blob/v0.7.0/pkg/backends/vault.go#L40) there were no versioned secrets support for Vault so your input (`avp.kubernetes.io/secret-version: "latest"`) doesn't have any effect.

If you look at the Vault client code in version [1.12.0](https://github.com/argoproj-labs/argocd-vault-plugin/blob/v1.12.0/pkg/backends/vault.go#L50) version parameters are respected and if you don't specify any, it takes the latest one.

In short, `latest` doesn't mean anything to the Vault therefore you only need to specify it if you really need to use a specific version. If you want the latest one, just don't add the version.

**Comment by gogovan-vincentngai** [comment]
Yes i also got this answer after few test 
Anyway this argocd-vault-plugin is awesome
